lib/player.js

```diff
@@ -6512,7 +6512,10 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     // Because this.video_ may not be set when the config is built, the default
     // TextDisplay factory must capture a reference to "this".
     config.textDisplayFactory = () => {
-      if (this.videoContainer_) {
+      // On devices where the Fullscreen API is not available we prefer
+      // SimpleTextDisplayer because it works with the Fullscreen API of the
+      // video element itself.
+      if (this.videoContainer_ && document.fullscreenEnabled) {
```

Comment:
I think this should be a little more subtle.  If you had a device without a fullscreen API of any kind, we would still want UITextDisplayer.  The issue is that we end up on the WebKit-specific video-only fullscreen method, which breaks UITextDisplayer.

I think what we really want is something like `this.videoContainer_ && (shouldUseDocumentFullscreen(this.config_) || !isFullScreenSupported())`, where `shouldUseDocumentFullscreen` and `isFullScreenSupported` are extracted from ui/controls.js and made static.

Is this too subtle/complex?

Reply:
Ok, I’ll do it tomorrow!